sbpy/utils/dastcom5.py

```diff
@@ -0,0 +1,575 @@
+        f.seek(835, os.SEEK_SET)
+        data = np.fromfile(f, dtype=AST_DTYPE)
+    return data
+
+
+def comet_db():
+    """Return complete DASTCOM5 comet database.
+
+    Returns
+    -------
+    database : numpy.ndarray
+        Database with custom dtype.
+
+    """
+    with open(COM_DB_PATH, "rb") as f:
+        f.seek(976, os.SEEK_SET)
+        data = np.fromfile(f, dtype=COM_DTYPE)
+    return data
+
+
+def orbit_from_name(name):
+    """Return :py:class:`~astropy.table.Table` given a name.
+
+    Retrieve info from JPL DASTCOM5 database.
+
+    Parameters
+    ----------
+    name : str
+        NEO name.
+
+    Returns
+    -------
+    Table : ~astropy.table.Table
+        Near Earth Asteroid/Comet orbit parameters, all stacked.
+
+    """
+    records = record_from_name(name)
+    orbits = [orbit_from_record(rec) for rec in records]
+    tbl = Table(orbits)
+    return tbl
+
+
+def orbit_from_record(record):
+    """Return :py:class:`~astropy.table.Table` given a record.
+
+        Retrieve info from JPL DASTCOM5 database.
+
+        Parameters
+        ----------
+        record : int
+            Object record.
+
+        Returns
+        -------
+        Table : ~astropy.table.Table
+            Near Earth Asteroid/Comet orbit parameters.
+
+    """
+    body_data = read_record(record)
+    a = body_data["A"].item()
+    ecc = body_data["EC"].item()
+    inc = body_data["IN"].item()
+    raan = body_data["OM"].item()
+    argp = body_data["W"].item()
+    m = body_data["MA"].item()
+    epoch = Time(body_data["EPOCH"].item(), format="jd", scale="tdb")
+    column2 = (record, a, ecc, inc, raan, argp, m, epoch)
+    column1 = ("record", "a", "ecc", "inc", "raan", "argp", "m", "EPOCH")
+    data = Table(rows=[column1, column2])
+
+    return data
+
+
+def record_from_name(name):
+    """Search `dastcom.idx` and return logical records that
+    match a given string.
+
+    Body name, SPK-ID, or alternative designations can be used.
+
+    Parameters
+    ----------
+    name : str
+        Body name.
+
+    Returns
+    -------
+    records : list (int)
+        DASTCOM5 database logical records matching str.
+
+    """
+    records = []
+    lines = string_record_from_name(name)
+    for line in lines:
+        records.append(int(line[:6].lstrip()))
+    return records
+
+
+def string_record_from_name(name):
+    """Search `dastcom.idx` and return body full record.
+
+    Search DASTCOM5 index and return body records that match string,
+    containing logical record, name, alternative designations, SPK-ID, etc.
+
+    Parameters
+    ----------
+    name : str
+        Body name.
+
+    Returns
+    -------
+    lines: list(str)
+        Body records
+    """
+
+    idx_path = os.path.join(DBS_LOCAL_PATH, "dastcom.idx")
+    lines = []
+    with open(idx_path, "r") as inF:
+        for line in inF:
+            if re.search(r"\b" + name.casefold() + r"\b", line.casefold()):
+                lines.append(line)
+    return lines
```

Comment:
please turn this output into a `sbpy.data.DataClass` object with corresponding column names.